template.yaml

```diff
@@ -1384,6 +1299,252 @@ Resources:
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: kms:*
+            Resource: "*"
+          - Effect: Allow
+            Principal:
+              Service: !Sub logs.${AWS::Region}.amazonaws.com
+            Action:
+              - kms:Encrypt*
+              - kms:Decrypt*
+              - kms:ReEncrypt*
+              - kms:GenerateDataKey*
+              - kms:Describe*
+            Resource: "*"
+            Condition:
+              ArnLike:
+                kms:EncryptionContext:aws:logs:arn: !Sub arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*
+          - Effect: Allow
+            Principal:
+              AWS: arn:aws:iam::731252101191:role/GDS-SQS-Role-nonprod
+            Resource: "*"
+            Action:
+              - kms:GenerateDataKey
+              - kms:Decrypt
+
+  DwpOldFormatDeathKmsKeyAlias:
+    Condition: IsIntegration
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub alias/${Environment}${Developer}/DwpOldFormatDeathKmsKey
+      TargetKeyId: !Ref DwpOldFormatDeathKmsKey
+
+  DwpOldFormatDeathSnsToSqsPolicy:
+    Condition: IsIntegration
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: Allow SNS publish to SQS
+            Effect: Allow
+            Principal:
+              Service: sns.amazonaws.com
+            Resource: "*"
+            Action: SQS:SendMessage
+            Condition:
+              ArnEquals:
+                aws:SourceArn: !Ref DeathDistributionTopic
+      Queues:
+        - !Ref DwpOldFormatDeathMinimisationQueue
+
+  DwpOldFormatDeathDistributionDlq:
+    Condition: IsIntegration
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${Environment}${Developer}-dwp-old-format-death-distribution-dlq
+      KmsMasterKeyId: !Ref MainKmsKey
+
+  DwpOldFormatDeathDistributionTopicSubscription:
+    Condition: IsIntegration
+    Type: AWS::SNS::Subscription
+    Properties:
+      Protocol: sqs
+      Endpoint: !GetAtt DwpOldFormatDeathMinimisationQueue.Arn
+      TopicArn: !GetAtt DeathDistributionTopic.TopicArn
+      RawMessageDelivery: true
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt DwpOldFormatDeathDistributionDlq.Arn
+
+  DwpOldFormatDeathMinimisationQueue:
+    Condition: IsIntegration
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${Environment}${Developer}-dwp-old-format-death-minimisation
+      KmsMasterKeyId: !Ref MainKmsKey
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt DwpOldFormatDeathMinimisationDlq.Arn
+        maxReceiveCount: 5
+
+  DwpOldFormatDeathMinimisationDlq:
+    Condition: IsIntegration
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${Environment}${Developer}-dwp-old-format-death-minimisation-dlq
+      KmsMasterKeyId: !Ref MainKmsKey
+
+  DwpOldFormatDeathDeliveryQueue:
```

Comment:
Is there a reason we've changed these names to get rid of 'dev'?

Reply:
It needs to be processed separately as otherwise it would consume the (new format) messages off the delivery queue in order to convert them to the old format - we want both new and old formats available (I think that's why)

Reply:
sounds good!